dimod/generators/wireless.py

```diff
@@ -0,0 +1,985 @@
+    Constructs coefficients for the form
+    :math:`O(v) = v^{\dagger} J v - 2 \Re(h^{\dagger} v) + \\text{offset}`.
+
+    Args:
+        y: Received signal as a NumPy column vector of complex or real values.
+
+        F: Wireless channel as an :math:`i \\times j` NumPy matrix of complex
+            values, where :math:`i` rows correspond to :math:`y_i` receivers
+            and :math:`j` columns correspond to :math:`v_i` transmitted symbols.
+
+    Returns:
+        Three tuple of offset, as a real scalar, linear biases :math:`h`, as a dense
+        real vector, and quadratic interactions, :math:`J`, as a dense real symmetric
+        matrix.
+    """
+
+    if len(F.shape) != 2 or F.shape[0] != y.shape[0]:
+        raise ValueError("F should have shape (n, m) for some m, n "
+                         "and n should equal y.shape[1];"
+                         f" given: {F.shape}, n={y.shape[1]}")
+
+    offset = np.matmul(y.imag.T, y.imag) + np.matmul(y.real.T, y.real)
+    h = -2 * np.matmul(F.T.conj(), y)    # Be careful with interpretation!
+    J = np.matmul(F.T.conj(), F)
+
+    return offset, h, J
+
+def _real_quadratic_form(h, J, modulation=None):
+    """Separate real and imaginary parts of quadratic form.
+
+    Unwraps objective function on complex variables as an objective function of
+    concatenated real variables, first the real and then the imaginary part.
+
+    Args:
+        h: Linear biases as a dense NumPy vector.
+
+        J: Quadratic interactions as a dense symmetric matrix.
+
+        modulation: Modulation. Supported values are 'BPSK', 'QPSK', '16QAM', 
+            '64QAM', and '256QAM'.
+
+    Returns:
+        Two-tuple of linear biases, :math:`h`, as a NumPy real vector with any
+        imaginary part following the real part, and quadratic interactions,
+        :math:`J`, as a real matrix with any imaginary part moved to above and
+        below the diagonal.
+    """
+    # Here, for BPSK F-induced complex parts of h and J are discarded:
+    # Given y = F x + nu, for independent and identically distributed channel F
+    # and complex noise nu, the system of equations defined by the real part is
+    # sufficient to define the canonical decoding problem.
+    # In essence, rotate y to the eigenbasis of F, throw away the orthogonal noise
+    # (the complex problem as the real part with suitable adjustment factor 2 to
+    # signal to noise ratio: F^{-1}*y = I*x + F^{-1}*nu)
+    # JR: revisit and prove
+
+    if modulation and modulation not in mod_config:
+        raise ValueError(f"Unsupported modulation: {modulation}")
+
+    if modulation != 'BPSK' and (np.iscomplex(h).any() or np.iscomplex(J).any()):
+        hR = np.concatenate((h.real, h.imag), axis=0)
+        JR = np.concatenate((np.concatenate((J.real, J.imag), axis=0),
+                             np.concatenate((J.imag.T, J.real), axis=0)),
+                             axis=1)
+        return hR, JR
+    else:
+        return h.real, J.real
```

Comment:
Very much a nit but....
```suggestion
    return h.real, J.real
```
I am reasonably convinced by the [LLVM Style Guide](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return)'s arguments on this sort of flow.